api/v1alpha1/hardwareevent_webhook.go

```diff
@@ -62,26 +72,53 @@ func (r *HardwareEvent) ValidateCreate() error {
 		return fmt.Errorf("only one Hardware Event instance is supported at this time")
 	}
 
-	if r.Spec.TransportHost == "" {
-		return fmt.Errorf("transport URL is required field in Hardware instance spec")
-	}
-	return nil
+	return r.validate()
 }
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (r *HardwareEvent) ValidateUpdate(old runtime.Object) error {
 	hardwareeventlog.Info("validate update", "name", r.Name)
-
-	if r.Spec.TransportHost == "" {
-		return fmt.Errorf("transport URL is required field in Hardware instance spec")
-	}
-	return nil
+	return r.validate()
 }
 
 // ValidateDelete implements webhook.Validator so a webhook will be registered for the type
 func (r *HardwareEvent) ValidateDelete() error {
 	hardwareeventlog.Info("validate delete", "name", r.Name)
+	return nil
+}
+
+func (r *HardwareEvent) validate() error {
 
-	// TODO(user): fill in your validation logic upon object deletion.
+	eventConfig := r.Spec
+	transportUrl, err := url.Parse(eventConfig.TransportHost)
+	if eventConfig.TransportHost == "" || err != nil {
+		hardwareeventlog.Info("transportHost is not valid, proceed as", "transportHost", DefaultTransportHost)
+	}
+
+	if eventConfig.TransportHost == "" || transportUrl.Scheme != AmqScheme {
```

Comment:
any reason for this to be a negative check like:
`transportUrl.Scheme != AmqScheme`
should it be checking for `transportUrl.Scheme == HttpScheme`

Reply:
the reason is we allow user to skip "http://" for http transport, for example
transportHost: "hw-event-publisher-service.openshift-bare-metal-events.svc.cluster.local:9043"
transportHost: "localhost"